app/models/user.rb

```diff
@@ -470,6 +470,11 @@ def coauthored_works
     return @coauthored_works
   end
 
+  # checks that only collections a user has created are deleted.
+  def sole_owned_collections
+    self.collections.to_a.delete_if { |collection| !(collection.all_owners - pseuds).empty? }
```

Comment:
I think using this would be more efficient: `self.owned_collections.select { |collection| collection.all_owners.count == 1 }`

Reply:
I'm sure if your statement is checking for the same thing mine is. I'd rather keep it as is.

Reply:
Yeah, I think @smclairecarden is right and these are two different checks. It looks like `all_owners` returns a list of pseuds, and it's possible for two pseuds belonging to the same user to be owners of a collection.